eldoc-box.el

```diff
@@ -578,6 +578,62 @@ display the docs in echo area depending on
     (when (eldoc-box--eldoc-message-function "%s" doc)
       (eldoc-display-in-echo-area docs interactive))))
 
+(defun eldoc-box--visible-p (&optional frame)
+  "Return non-nil if FRAME is visible."
+  (let ((frame (or frame eldoc-box--frame)))
+    (and (frame-live-p frame) (frame-visible-p frame))))
+
+(defun eldoc-box-scroll-up (&optional n)
+  "Scroll text of eldoc-box window upward N lines."
+  (interactive "p")
+  (if (eldoc-box--visible-p)
+      (with-selected-frame eldoc-box--frame
+        (with-current-buffer eldoc-box--buffer
+          (scroll-up n)))
+    (scroll-other-window n)))
+
+(defun eldoc-box-scroll-down (&optional n)
+  "Scroll text of eldoc-box window down N lines."
+  (interactive "p")
+  (eldoc-box-scroll-up (- (or n 1))))
+
+(defun eldoc-box-end (&optional n)
+  "Scroll text of eldoc-box window to the end.
+
+With numeric arg N, put window N/10 of the way from the end."
+  (interactive "P")
+  (if (eldoc-box--visible-p)
+      (with-selected-frame eldoc-box--frame
+        (with-current-buffer eldoc-box--buffer
+          (with-no-warnings
```

Comment:
I'd use condition-case and catch the specific error. I assume it's end-of-buffer in this case?

Reply:
I didn't see why we need use condition-case here.

All this function do is go to the eldoc-box's end. If error happens, we needn't handle it.

Reply:
Sure, that's fine.